### PR TITLE
Implement rubocop

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
@ekohl if you want to have rubocop here I can proceed and fix the violations

File: puppet_metadata.gemspec [hunk 1]
```diff
@@ -14,6 +14,7 @@ Gem::Specification.new do |s|
   s.files = Dir['lib/**/*.rb']
   s.extra_rdoc_files = ['README.md']
   s.rdoc_options << '--main' << 'README.md'
+  s.required_ruby_version = '>= 2.4.0'
```
Comment (ekohl) [inline puppet_metadata.gemspec:17]: I'd prefer an upper bound `< 4`

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -8,6 +8,17 @@ env:
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Ruby 3.1
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.1'
+          bundler-cache: true
+      - name: Run RuboCop
+        run: bundle exec rake rubocop
   test:
```
Comment (ekohl) [inline .github/workflows/test.yml:22]: You could make this depend on the rubocop step so you don't run tests unless Rubocop passes.

**Review by ekohl — COMMENTED** [review]
:+1: on adding Rubocop.

File: puppet_metadata.gemspec (hunk 1 above)
Comment (bastelfreak) [reply puppet_metadata.gemspec]: added the upper bound. I set 2.4.0 as lowest version because that's what currently CI runs on.

File: .github/workflows/test.yml (hunk 2 above)
Comment (ekohl) [reply .github/workflows/test.yml:22]: The test suite in this module is fast (< 15 seconds for a full run) so I guess it's fine.

File: .rubocop_todo.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
+# This configuration was generated by
```
Comment (ekohl) [inline .rubocop_todo.yml:1]: I'd like to avoid a large TODO file. Any reason we can't get these resolved?

File: lib/puppet_metadata/beaker.rb [hunk 4]
```diff
@@ -8,17 +10,17 @@ class Beaker
     # PIDFile parameter
     PIDFILE_COMPATIBLE_IMAGES = {
       'CentOS' => {
-        '7' => 'centos:7.6.1810',
+        '7' => 'centos:7.6.1810'
```
Comment (ekohl) [inline lib/puppet_metadata/beaker.rb:13]: Can we set the style to keep trailing commas instead?

**Review by ekohl — CHANGES_REQUESTED** [review]

File: .rubocop_todo.yml (hunk 3 above)
Comment (bastelfreak) [reply .rubocop_todo.yml:1]: available time limits this :D 
Fixing some of them would be a breaking change (like renaming methods) or take a lot of time (like max black/class length). I think this is still a good improvement because it will at least lint new code and prevent the codebase from getting worse.

File: .github/workflows/test.yml (hunk 2 above)
Comment (bastelfreak) [reply .github/workflows/test.yml:22]: updated it

File: lib/puppet_metadata/github_actions.rb [hunk 5]
```diff
@@ -56,9 +55,7 @@ def beaker_os_releases
         else
           releases&.each do |release|
             majors.each do |puppet_version|
-              if AIO.has_aio_build?(os, release, puppet_version[:value])
-                yield [os, release, puppet_version]
-              end
+              yield [os, release, puppet_version] if AIO.has_aio_build?(os, release, puppet_version[:value])
```
Comment (ekohl) [inline lib/puppet_metadata/github_actions.rb:58]: I think we should disable the guard cop here because we also disable the line length cop, in which case the guard cop is bad (IMHO). This very long line is hard to parse mentally.

File: .rubocop_todo.yml [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+# Configuration parameters: Severity, Include.
+# Include: **/*.gemspec
+Gemspec/DuplicatedAssignment:
+  Exclude:
+    - 'puppet_metadata.gemspec'
+
+# Offense count: 1
+# Configuration parameters: Severity, Include.
+# Include: **/*.gemspec
+Gemspec/RequiredRubyVersion:
```
Comment (ekohl) [inline .rubocop_todo.yml:19]: I think this was solved. Perhaps the TODO file needs to be generated?

Comment (bastelfreak) [reply .rubocop_todo.yml]: done

File: lib/puppet_metadata/github_actions.rb (hunk 5 above)
Comment (bastelfreak) [reply lib/puppet_metadata/github_actions.rb]: done
